### DISCUSSION TITLE
Solution to "could not decrypt payloads".  Advice on how to change MTU and MSS settings.

### DISCUSSION THREAD

**Discussion by Rentaro-K** [strongswan/strongswan#1395] [Q&A]
■ Issue.
We have installed strongSwan on Ubuntu 22.04 in an AWS environment.
We have performed the configuration work and tried to establish an IPsec-VPN connection with the on-premises device, but it failed.
The status log for this is as follows.
We have confirmed that the PSKs match, so we believe this is due to packet corruption

> Nov 04 06:18:11 ip-172-32-29-108 charon[23863]: 07[ENC] invalid HASH_V1 payload length, decryption failed?
> Nov 04 06:18:11 ip-172-32-29-108 charon[23863]: 07[ENC] could not decrypt payloads
> Nov 04 06:18:11 ip-172-32-29-108 charon[23863]: 07[IKE] message parsing failed
> Nov 04 06:18:11 ip-172-32-29-108 charon[23863]: 07[IKE] ignore malformed INFORMATIONAL request
> Nov 04 06:18:11 ip-172-32-29-108 charon[23863]: 07[IKE] INFORMATIONAL_V1 request with message ID 4267074650 processing failed

■ What we did.
Packet corruption may be resolved by changing MTU or MSS settings.
Investigation revealed that the MTU needs to be set below a certain value in the AWS environment.
The following configuration changes have been made to Ubuntu, but the problem has not been resolved.

> ubuntu@ip-172-32-29-108:~$ sudo ip link set ens5 mtu 1399

Although not yet implemented, the following MSS configuration changes will also be made in the future.

> iptables -t mangle -A FORWARD -m policy --pol ipsec --dir in -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
> iptables -t mangle -A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
> 

・Reference.
https://docs.aws.amazon.com/vpn/latest/s2svpn/your-cgw.html

MTU in AWS < 1399
https://docs.aws.amazon.com/vpn/latest/s2svpn/your-cgw.html#cgw-best-practice

■Question.
・What are the possible causes of the above logs other than "PSK mismatch" or "incorrect MTU/MSS settings"?
・Is the MTU and MSS configuration change work valid only for Ubuntu?
　We are concerned that the behaviour may not change properly without configuration change work on strongSwan as well as Ubuntu.
　In that case, it would be very helpful if you could tell us how to change MTU and MSS on strongSwan.
・We believe that packet corruption is caused by packet fragmentation.
　Could the situation be improved by setting "fragmentation = no" in ipsec.conf?
　
　
■Reference
my syslog
[ss_syslog.txt](https://github.com/strongswan/strongswan/files/9936017/ss_syslog.txt)

My ipsec.conf

> conn %default
> 
> conn xxxx
>         authby=psk
>         dpdaction=restart
>         dpddelay=60s
>         dpdtimeout=240s
>         auto=start
>         leftsubnet=AAA.AAA.AAA.AAA/16
>         right=BBB.BBB.BBB.BBB
>         rightsubnet=10.10.20.0/24
>         keyexchange=ikev1
>         ikelifetime=28800s
>         ike=aes256-sha1-modp2048
>         aggressive=no
>         type=tunnel
>         lifetime=3600s
>         esp=aes256-sha1-modp2048
>         leftid=%any
>         rightid=%any
>         rekey=yes

**Comment by Rentaro-K** [comment]
Added a sysylog.
[ss_syslog.txt](https://github.com/strongswan/strongswan/files/9936028/ss_syslog.txt)

Sorry for the additional questions.

Reading the syslog, I see the following log.

"Nov 4 06:29:23 ip-172-32-29-108 ipsec[24020]: 14[IKE] IKE_SA xxxx[2] established between AAA.AAA.AAA.AAA[AAA.AAA.AAA]. .zzz.zzz.zzz.zzz.zzz[zzz.zzz.zzz.zzz]"

Reading only this log, it appears as if the IKE has succeeded and the IPsec-VPN connection has been opened.

An additional question is at what stage of the key exchange am I failing?

**Reply by tobiasbrunner** [reply]
Main mode seems to get completed successfully (at least as responder, initiating fails as there is apparently no PSK that matches your config). No idea why it fails as responder afterwards. The INFORMATIONAL that can't be decrypted could be an error message, contain an INITIAL_CONTACT notify or it could be a Mode Config request if virtual IPs are used or for XAuth authentication. You'd have to check on the other end what's going on.

Anyway, you should really be using IKEv2 instead of IKEv1.

**Reply by Rentaro-K** [reply]
Thank you for your reply.

I would like to collect more detailed logs when starting strongswan.
Is there a better way to check the status and logs other than checking syslog at debug level?

**Reply by tobiasbrunner** [reply]
> I would like to collect more detailed logs when starting strongswan.

What for? It's not a problem on the strongSwan side.

**Reply by Rentaro-K** [reply]
Thank you for your reply.

We understand that there seems to be no problem with strongSwan.

Apart from that, I asked the question out of technical curiosity, as I thought it would be useful for future travel shooting if I could read more detailed logs.

**Reply by tobiasbrunner** [reply]
See #196 for recommended log settings for config issues.

**Comment by Rentaro-K** [comment]
Thank you for your reply.

Lastly, a confirmation.
Am I correct in understanding that when changing MTU and MSS settings, we are only operating on Ubuntu and not on strongswan?


> ・Is the MTU and MSS configuration change work valid only for Ubuntu?
> 　We are concerned that the behaviour may not change properly without configuration change work on strongSwan as well as Ubuntu.

**Reply by tobiasbrunner** [reply]
Please see [here](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_mtumss_issues).
